src/service/searchKnowledgeStore.js

```diff
@@ -27,10 +28,18 @@ async function searchKnowledgeStore(query) {
   });
 
   const filteredResults = [];
+  const highlights = new Set();
   for await (const result of searchResults.results) {
     // https://learn.microsoft.com/en-us/azure/search/search-pagination-page-layout#order-by-the-semantic-reranker
     // drop anything below 2 as they generally aren't that relevant
     if (result.rerankerScore && result.rerankerScore > 2) {
+      // duplicate results can happen in the search results, so we need to filter them out
+      const highlight = extractKnowledgeStoreHighlight(result);
```

Comment:
I found certain searches could cause similar results to come up with the exact same highlights which looked bad